### PR TITLE
general improvements to http api

### PR DESCRIPTION
**Pull request by dyc3**

### CODE REVIEW

File: http/api.go [hunk 1]
```diff
@@ -157,7 +157,10 @@ func HandleBattleSimulate(w http.ResponseWriter, r *http.Request) {
 
 	bytes, err := json.Marshal(results)
 	if err != nil {
-		log.Fatalf("Failed to marshal into JSON: %s", err)
+		log.Printf("Failed to marshal into JSON: %s", err)
```
Comment (A-very-Cunning-ham) [inline http/api.go:160]: is there a reason this isn't a Panicf like the similar JSON error in agent.go?

Comment (dyc3) [reply http/api.go:160]: The panic in agent.go is to get it to leave the function and fail. The print here is to let the server respond to the request.

Comment (A-very-Cunning-ham) [reply http/api.go:160]: makes sense!

**Review by A-very-Cunning-ham — APPROVED** [review]
looking good

**Pull request merged**